script/grblcon.py

```diff
@@ -0,0 +1,527 @@
+
+    def _clear_screen(self):
+        """ Resets the text display area
+
+        :returns: None
+        :rtype: None
+
+        """
+
+        self._text_display.set_text('\n' * self.rows)
+
+    def quit(self):
+        """ Cleans up and tries to quit the program
+
+        :returns: None
+        :rtype: None
+
+        """
+        self._close_serial()
+        raise urwid.ExitMainLoop
+
+    def _handle_clear(self, *args):  # pylint: disable=unused-argument
+        self._clear_screen()
+
+    def _handle_help(self, *args):  # pylint: disable=unused-argument
+        help_str = '\n'.join([
+            'Available Commands:',
+            '-------------------',
+        ])
+
+        esc_handlers = [x.replace('_handle_', '') for x
+                        in dir(self)
+                        if x.startswith('_handle')]
+
+        esc_handlers.append('\\')
+
+        self._update_text_display(help_str)
+        max_cmd_len = max([len(x) for x in self._command_help])
+        for handler in esc_handlers:
+            help_str = '{} - {}'.format(handler.ljust(max_cmd_len),
+                                        self._command_help[handler])
+            self._update_text_display(help_str)
+
+        help_footer = ('\nNote: Commands are listed as '
+                       '{command} - /<arity> (args) [help]\n'
+                       'args should be space separated!')
+        self._update_text_display(help_footer)
+
+    def _handle_quit(self, *args):  # pylint: disable=unused-argument
+        self.quit()
+
+    def _handle_toggle_quiet(self, *args):  # pylint: disable=unused-argument
+        self._quiet = not self._quiet
+
+    def _handle_close(self, *args):  # pylint: disable=unused-argument
+        self._log_info('Closing GRBL Connection')
+        self._close_serial()
+
+    def _handle_open(self, *args):  # pylint: disable=unused-argument
+        port, baud = args[0]
+        self._log_info('Opening GRBL connection @ {}::{}'.format(port, baud))
+        self._open_port(port, int(baud))
+
+    def _handle_log(self, *args):  # pylint: disable=unused-argument
+        self._delimit_display()
+        self._update_text_display(self._event_log)
+
+    def _update_text_display(self, text):
+        """ Updates the central display widget with an additional line of text
+
+        :param text: a single line of text to insert at the bottom
+        :returns: None
+        :rtype: None
+
+        """
+
+        new_text = _update_text(self.rows - 2,  # Account for the header/footer
+                                self._text_display.text,
+                                text)
+
+        self._text_display.set_text(new_text)
+
+    def _process_escape_seq(self, text):
+        # Strip off the escape character
+        text = text[1:]
+
+        self._update_text_display(text)
+        if self._port:
+            self._port.write(bytes(text + '\n', 'ascii'))
+
+    def _process_cmd(self, text):
+        tokens = text.split(' ')
```

Comment:
The default argument to `.split` splits on whitespace. What you've got splits on individual spaces, so that an extra space will make this longer than you might intend. Just use the default.